runtime/src/migration.rs

```diff
@@ -0,0 +1,99 @@
+pub mod session_keys {
+	use frame_support::{pallet_prelude::Weight, traits::OnRuntimeUpgrade};
+	use sp_runtime::impl_opaque_keys;
+	use sp_std::vec::Vec;
+
+	use polkadex_primitives::AccountId;
+
+	use crate::{AuthorityDiscovery, Babe, Grandpa, ImOnline, Runtime, SessionKeys};
+
+	impl_opaque_keys! {
+		pub struct SessionKeysV4 {
+			pub grandpa: Grandpa,
+			pub babe: Babe,
+			pub im_online: ImOnline,
+			pub authority_discovery: AuthorityDiscovery,
+		}
+	}
+
+	pub struct MigrateToV5;
+
+	impl MigrateToV5 {
+		fn run() -> Weight {
+			let translate_fn = |keys: Option<Vec<(AccountId, SessionKeysV4)>>| {
+				let mut new_keys: Vec<(AccountId, SessionKeys)> = Vec::new();
+				if let Some(keys) = keys {
+					for (validator, keys) in keys {
+						let new_key = (
+							validator,
+							SessionKeys {
+								grandpa: keys.grandpa,
+								babe: keys.babe,
+								im_online: keys.im_online,
+								authority_discovery: keys.authority_discovery,
+								orderbook: match [0u8; 96].as_ref().try_into() {
+									Ok(ob) => ob,
+									Err(_) => return None,
+								}, // Set empty public key
+								thea: match [0u8; 96].as_ref().try_into() {
+									Ok(thea) => thea,
+									Err(_) => return None,
+								},
+							},
+						);
+						log::info!(target:"migration","Migrated session key: {:?}",new_key);
+						new_keys.push(new_key);
+					}
+				}
+				Some(new_keys)
+			};
+
+			if pallet_session::QueuedKeys::<Runtime>::translate::<
+				Vec<(AccountId, SessionKeysV4)>,
+				_,
+			>(translate_fn).is_err() {
+				log::error!(target:"migration","Storage type cannot be interpreted as the Vec<(AccountId, SessionKeysV4)>")
+			}
+
+			pallet_session::NextKeys::<Runtime>::translate::<SessionKeysV4, _>(|_, old_keys| {
+				Some(SessionKeys {
+					grandpa: old_keys.grandpa,
+					babe: old_keys.babe,
+					im_online: old_keys.im_online,
+					authority_discovery: old_keys.authority_discovery,
+					orderbook: match [0u8; 96].as_ref().try_into() {
+						Ok(ob) => ob,
+						Err(_) => return None,
+					}, // Set empty public key
+					thea: match [0u8; 96].as_ref().try_into() {
+						Ok(thea) => thea,
+						Err(_) => return None,
+					},
+				})
+			});
+			Weight::zero()
```

Comment:
Is this supposed to be updated at some point? (If so - please add a todo). Otherwise, perhaps  this method could be void and default `Weight::zero()` could be returned in `on_runtime_upgrade` method.

Reply:
The fact that upgrade is executed with unchecked weight allow us to get away with it.